### PR TITLE
Add template variables to dataset card template

### PR DESCRIPTION
**Pull request by nateraw**
This PR adds jinja template variables to dataset card so you can more easily programmatically update it if you wish. There currently doesn't exist a great way to test this - maybe we should think about that in the future.

### CODE REVIEW

File: tests/test_repocard.py [hunk 1]
```diff
@@ -743,6 +743,31 @@ def test_dataset_card_from_default_template(self):
...
+            repo_url="https://github.com/huggingface/huggingface_hub",
+            paper_url="https://arxiv.org/pdf/1910.03771.pdf",
+            point_of_contact="https://huggingface.co/nateraw",
+            dataset_summary=(
+                "This is a test dataset card to check if the template variables "
+                "in the dataset card template are working."
+            ),
+        )
+        self.assertTrue(
+            card.text.strip().startswith("# Dataset Card for My Cool Dataset")
```
Comment (nateraw) [inline tests/test_repocard.py:767]: should probably do a regex check for something later in the file here

**Review by Wauplin — APPROVED** [review]
Looks good to me. I'll let @lhoestq review as well and then we can merge :)

**Review by lhoestq — APPROVED** [review]
Thanks ! note that for vision or audio datasets we may suggest another template

**Comment by Wauplin** [comment]
> Thanks ! note that for vision or audio datasets we may suggest another template

I am also linking [this discussion](https://huggingface.slack.com/archives/C025H22A3ND/p1663706093539299) (internal link) that could be of interest.
And thanks for the review, I'm closing the PR :)

**Pull request merged**